tests/integration/test_arg_reduce.py

```diff
@@ -20,26 +20,145 @@
+        ndim = 2
+        shape = (5,) * ndim
+        in_num = np.random.random(shape)
+        msg = r"axis must be smaller than array.ndim and cannot be negative"
+        for fn in ("argmax", "argmin"):
+            fn_num = getattr(num, fn)
+            with pytest.raises(ValueError, match=msg):
+                fn_num(in_num, axis=-(ndim + 1))
+
+    def test_out_float(self):
+        shape = (5,) * 3
+        in_num = np.random.random(shape)
+        msg = r"output array must have int64 dtype"
+        for fn in ("argmax", "argmin"):
+            fn_num = getattr(num, fn)
+            res_out = np.random.random(size=(1, 5, 5))
+            with pytest.raises(ValueError, match=msg):
+                fn_num(in_num, out=res_out)
+
+    def test_out_shape_mismatch(self):
+        ndim = 3
+        shape = (5,) * ndim
+        in_num = np.random.random(shape)
+        msg = r"the output shape mismatch"
+        for fn in ("argmax", "argmin"):
+            fn_num = getattr(num, fn)
+            res_out = np.random.randint(1, 10, size=shape)
+            with pytest.raises(ValueError, match=msg):
+                fn_num(in_num, out=res_out)
+
+
+class TestArgMaxAndArgMin:
+    """
+    These are positive cases compared with numpy
+    """
+
+    @pytest.mark.parametrize("ndim", range(LEGATE_MAX_DIM + 1))
+    @pytest.mark.parametrize("keepdims", [True, False])
+    def test_argmax_and_argmin_basic(self, ndim, keepdims):
+        shape = (5,) * ndim
+        in_np = np.random.random(shape)
+        in_num = num.array(in_np)
+        for fn in ("argmax", "argmin"):
+            fn_np = getattr(np, fn)
+            fn_num = getattr(num, fn)
+            assert np.array_equal(
+                fn_np(in_np, keepdims=keepdims),
+                fn_num(in_num, keepdims=keepdims),
+            )
+
+    @pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+    @pytest.mark.parametrize("keepdims", [True, False])
+    def test_argmax_and_argmin_axis(self, ndim, keepdims):
+        shape = (5,) * ndim
+        in_np = np.random.random(shape)
+        in_num = num.array(in_np)
+        for fn in ("argmax", "argmin"):
+            fn_np = getattr(np, fn)
+            fn_num = getattr(num, fn)
+            axis_list = list(range(in_num.ndim))
+            axis_list.append(-(ndim - 1))
+            for axis in axis_list:
+                out_np = fn_np(in_np, axis=axis, keepdims=keepdims)
+                out_num = fn_num(in_num, axis=axis, keepdims=keepdims)
+                assert np.array_equal(out_np, out_num)
+
+    @pytest.mark.parametrize("ndim", range(0, LEGATE_MAX_DIM + 1))
+    @pytest.mark.parametrize("keepdims", [True, False])
+    def test_argmax_and_argmin_out(self, ndim, keepdims):
+        shape = (5,) * ndim
+        in_np = np.random.random(shape)
+        in_num = num.array(in_np)
+        for fn in ("argmax", "argmin"):
+            fn_np = getattr(np, fn)
+            fn_num = getattr(num, fn)
+            if ndim == 0:
```

Comment:
If `ndim == 0` needs special treatment, then pull it into a separate test. Having large conditional blocks makes this test over-complicated.

Reply:
OK. Fix it.